packages/navigators/index.tsx

```diff
@@ -107,7 +107,9 @@ class AppWithNavigationState extends React.Component<Props> {
 
   onBackPress = () => {
     const { nav, navigation } = this.props;
-    if (nav.index === 0 && nav.routes[0].index === 0) {
+    if(nav.routes && nav.routes[1] && nav.routes[1].routeName === 'FriendDetail') {
```

Comment:
In some languages this could crash if nav.routes has 1 element, prob not in Javascript.
However better to check the array has length > 1 if you access nav.routes[1]